service/history/api/create_workflow_util.go

```diff
@@ -62,7 +63,7 @@ func NewWorkflowWithSignal(
 	runID string,
 	startRequest *historyservice.StartWorkflowExecutionRequest,
 	signalWithStartRequest *workflowservice.SignalWithStartWorkflowExecutionRequest,
-) (WorkflowContext, error) {
+) (WorkflowContext, *workflow.WorkflowTaskInfo, error) {
```

Comment:
I'm thinking of reverting this change and extracting the current task with `GetInFlightWorkflowTask`.